File: infrastructure/src/main/java/org/haas/infrastructure/persistence/screening/entity/ScreeningEntity.java
package org.haas.infrastructure.persistence.screening.entity;

import jakarta.persistence.*;
import lombok.*;
import org.haas.infrastructure.persistence.movie.entity.MovieEntity;
import org.haas.infrastructure.persistence.theater.entity.TheaterEntity;

import java.time.LocalDateTime;

@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Builder
@Getter
@Table(name = "screenings")
@Entity
public class ScreeningEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long screeningId;

    private LocalDateTime startTime;

    private LocalDateTime endTime;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "theater_id")
    private TheaterEntity theater;

}

File: data/src/main/kotlin/kr/spartacodingclub/cinema/data/entity/ScreeningEntity.kt
package kr.spartacodingclub.cinema.data.entity

import jakarta.persistence.*
import kr.spartacodingclub.cinema.core.domain.Screening
import java.time.LocalDateTime

@Entity
@Table(name = "screening")
class ScreeningEntity(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0L,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(
        name = "movie_id",
        foreignKey = ForeignKey(ConstraintMode.NO_CONSTRAINT)
    )
    val movie: MovieEntity,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(
        name = "theater_id",
        foreignKey = ForeignKey(ConstraintMode.NO_CONSTRAINT)
    )
    val theater: TheaterEntity,

    val startTime: LocalDateTime
) : BaseEntity() {
    fun toDomain() = Screening(
        id = id,
        movieId = movie.id,
        theater = theater.toDomain(),  // theater 엔티티를 도메인 객체로 변환
        startTime = startTime
    )
}
